### ISSUE TITLE
Add template tags to MPT v2 UI

### ISSUE THREAD

**Issue by milesmatthias** [spinnaker/spinnaker#4443]
### Issue Summary:

Running `spin pt save --file <template.json>` runs successfully, but listing pipeline templates (either through spin or the new UI) results in 3x the templates.

### Cloud Provider(s):

GCP

### Environment:

Spinnaker 1.14.0 on GKE

### Feature Area:

@spinnaker/gate

### Description:

I assume this has to do with the templates I'm using, I've tried several different templates and this behavior has been consistent.

### Steps to Reproduce:

1. Use `spin` to save a v2 pipeline template
2. View templates with `spin list` or in the UI

### Additional Details:

The templates I'm using have only helm bake stages in them, along with a pubsub trigger, some GCS artifacts, and some variables. I tried varying the number of each of those but no luck. Here's an example of a template that results in 3 listings for me:

```
{
  "id": "helm_bake",
  "schema": "v2",
  "protect": "false",
  "metadata": {
    "name": "Helm Bake template",
    "description": "Pipeline for helm bake",
    "owner": "owner@example.com",
    "scopes": ["global"]
  },
  "variables": [
    {
      "name": "appName",
      "description": "Name of your app",
      "defaultValue": "unknown",
      "type": "string"
    },
    {
      "name": "gitUrl",
      "description": "Git Repo URL",
      "defaultValue": "unknown",
      "type": "string"
    }
  ],
  "pipeline": {
    "expectedArtifacts": [
      {
        "defaultArtifact": {
          "id": "9c3cd6f9-83e7-4457-9d6a-b2fee7341628",
          "kind": "default.gcs",
          "name": "gs://deployment-artifacts/${templateVariables.appName}/charts/${templateVariables.appName}.tgz",
          "reference": "gs://deployment-artifacts/${templateVariables.appName}/charts/${templateVariables.appName}.tgz",
          "type": "gcs/object"
        },
        "displayName": "${templateVariables.appName}-chart",
        "id": "6bb25eb8-de81-4607-ab90-4baeb4366afb",
        "matchArtifact": {
          "id": "ca0214a5-4495-480d-a9e2-30bd80e18e46",
          "kind": "gcs",
          "name": "gs://deployment-artifacts/${templateVariables.appName}/charts/${templateVariables.appName}.tgz",
          "type": "gcs/object"
        },
        "useDefaultArtifact": true,
        "usePriorArtifact": false
      },
      {
        "defaultArtifact": {
          "id": "e6d599f4-4f48-48c1-b6c9-5464edba0d47",
          "kind": "default.gcs",
          "name": "gs://deployment-artifacts/${templateVariables.appName}/values/values.yaml",
          "reference": "gs://deployment-artifacts/${templateVariables.appName}/values/values.yaml",
          "type": "gcs/object"
        },
        "displayName": "values",
        "id": "d7dd1f97-b4ec-47f0-a9db-62cd09fddc19",
        "matchArtifact": {
          "id": "da9c7488-e215-4d32-a66d-c1afbafd5b36",
          "kind": "gcs",
          "name": "gs://deployment-artifacts/${templateVariables.appName}/values/values.yaml",
          "type": "gcs/object"
        },
        "useDefaultArtifact": true,
        "usePriorArtifact": false
      }
    ],
    "keepWaitingPipelines": false,
    "lastModifiedBy": "anonymous",
    "limitConcurrent": true,
    "locked": {
      "allowUnlockUi": false,
      "ui": true
    },
    "parameterConfig": [],
    "stages": [
      {
        "expectedArtifacts": [
          {
            "defaultArtifact": {
              "kind": "default.base64",
              "type": "embedded/base64"
            },
            "id": "466c5698-efcc-4bf0-af7a-6b8db8e07e7c",
            "matchArtifact": {
              "kind": "base64",
              "name": "${templateVariables.appName}-manifest",
              "type": "embedded/base64"
            },
            "useDefaultArtifact": false
          }
        ],
        "failOnFailedExpressions": true,
        "inputArtifacts": [
          {
            "account": "deployment-artifacts",
            "id": "6bb25eb8-de81-4607-ab90-4baeb4366afb"
          },
          {
            "account": "deployment-artifacts",
            "id": "19083e44-7081-41d2-9ab0-39212d1f8be4"
          }
        ],
        "name": "Bake (Manifest)",
        "namespace": "default",
        "outputName": "${templateVariables.appName}",
        "overrides": {},
        "refId": "1",
        "requisiteStageRefIds": [],
        "templateRenderer": "HELM2",
        "type": "bakeManifest"
      }
    ],
    "triggers": [
      {
        "attributeConstraints": {},
        "enabled": true,
        "payloadConstraints": {
          "appName": "${templateVariables.appName}",
          "gitUrl": "${templateVariables.gitUrl}"
        },
        "pubsubSystem": "google",
        "subscriptionName": "spinnaker-trigger",
        "type": "pubsub"
      }
    ]
  }
}
```

**Comment by jtk54** [comment]
We added tagging support in the v2 MPT API, so each `pt save` call saves

* The original template addressed by the id
* A copy of the template addressed by any provided tags or 'latest' addressed by 'id:<tag>'
* A content-addressable copy of the content addressed by 'id:<sha>'

Is the main issue here confusion or are there storage space constraints we're exceeding?

**Comment by milesmatthias** [comment]
~~Ah, thanks to @jtk54 for adding the labels. I do see in GCS (the front store for this instance) that there are 3 directories for each pipeline template.~~ Was just about to post, thanks for the clarification @jtk54.

The issue here is confusion for sure. Probably simply because MPT v2 is still early on?

I have lots of questions from the perspective of the template maintainer (managing tags, etc.), but the more immediate question is from the perspective of the person creating a pipeline based off of a template -- which template should I use? what happens when the template gets a new version/tag? 

The most immediate usage here is basing a pipeline on the latest version of a template, so you get the benefits when someone updates a template. Is it possible to see which of the 3 is the latest?

**Comment by jtk54** [comment]
This will eventually all be written up in docs, but there's a specific set of tags that [denote phases of the MPT lifecycle](https://github.com/spinnaker/front50/blob/master/front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java#L61). The idea is to have MPT consumers specify a tag they want to consume and be able to be auto-upgraded if a new MPT is pushed to the same tag. `latest` is indeed in the set of tags.

There will be UI features to know which MPT + tag a pipeline is using and support tags in templated pipeline creation.

**Comment by milesmatthias** [comment]
I guess if there isn't a ticket for the UI improvements to know which tag of a template you're using when clicking "Create Pipeline", this ticket can be renamed and be used for that purpose.

Since this isn't really a bug, just a half-baked feature that causes some confusion.

**Comment by louisjimenez** [comment]
I'm moving this into the [MPTv2 project](https://github.com/spinnaker/spinnaker/projects/7). The UI could benefit from surfacing versioning and tag information.

**Comment by louisjimenez** [comment]
Related to https://github.com/spinnaker/spin/issues/189

**Comment by louisjimenez** [comment]
I split this issue in two. The first [issue](https://github.com/spinnaker/spinnaker/issues/4561) removes the confusing triple records in favor of displaying and creating from "latest". That is merged and will be cherry picked onto 1.14. 

The second [issue](https://github.com/spinnaker/spinnaker/issues/4563) is for expanding on the UI for displaying templates and visually indicating versioning. That would give flexibility to choose which template version to reference while visually indicating the tags and digests a template corresponds with.
